src/sinks/nats.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{
+    config::{DataType, GenerateConfig, SinkConfig, SinkContext, SinkDescription},
+    emit,
+    event::Event,
+    internal_events::NatsEventSent,
+    sinks::util::StreamSink,
+};
+use async_trait::async_trait;
+use futures::{future, stream::BoxStream, FutureExt, StreamExt};
+use serde::{Deserialize, Serialize};
+
+/**
+ * Code dealing with the SinkConfig struct.
+ *
+ * DEV: Start with the bare minimum for now.
+ */
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NatsSinkConfig {
+    url: String,
+    subject: String,
+}
+
+inventory::submit! {
+    SinkDescription::new::<NatsSinkConfig>("nats")
+}
+
+impl GenerateConfig for NatsSinkConfig {}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "nats")]
+impl SinkConfig for NatsSinkConfig {
+    async fn build(
+        &self,
+        _cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        let sink = NatsSink::new(self.clone());
+        let healthcheck = future::ok(()).boxed();
+
+        Ok((super::VectorSink::Stream(Box::new(sink)), healthcheck))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "nats"
+    }
+}
+
+/**
+ * Code dealing with the Sink struct.
+ */
+
+#[derive(Clone)]
+pub struct NatsSink {
+    nc: nats::Connection,
+    subject: String,
```

Comment:
I think we should make this field templatable to let different messages flow to different topics. You can see an example of this in the `kafka` sink for the `topic` field:

https://github.com/timberio/vector/blob/2ee05ab058e4d19cb3b4fa7e0adfca92389c610a/src/sinks/kafka.rs#L135

Reply:
As best I could, I copied what Kafka had as far as the `Template` stuff goes